datalake/src/main/java/com/sequenceiq/datalake/flow/datalake/scale/handler/DatalakeHorizontalScaleServicesRollingRestartWaitHandler.java

```diff
@@ -0,0 +1,116 @@
+import com.sequenceiq.datalake.flow.datalake.scale.event.DatalakeHorizontalScaleFlowEvent;
+import com.sequenceiq.datalake.flow.datalake.scale.event.DatalakeHorizontalScaleFlowEvent.DatalakeHorizontalScaleFlowEventBuilder;
+import com.sequenceiq.flow.reactor.api.event.EventSender;
+import com.sequenceiq.flow.reactor.api.handler.EventSenderAwareHandler;
+
+@Component
+public class DatalakeHorizontalScaleServicesRollingRestartWaitHandler extends EventSenderAwareHandler<DatalakeHorizontalScaleFlowEvent> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DatalakeHorizontalScaleServicesRollingRestartWaitHandler.class);
+
+    private static final Long CONVERT_TO_MILLISECOND = 1000L;
+
+    @Value("${sdx.datalake.scale.sleeptime-sec:10}")
+    private int sleepTimeInSec;
+
+    @Value("${sdx.datalake.scale.timeout-sec:600}")
+    private int timeoutSec;
+
+    @Inject
+    private ClouderaManagerApiFactory clouderaManagerApiFactory;
+
+    @Inject
+    private ClouderaManagerProxiedClientFactory clouderaManagerProxiedClientFactory;
+
+    @Inject
+    private StackV4Endpoint stackV4Endpoint;
+
+    protected DatalakeHorizontalScaleServicesRollingRestartWaitHandler(EventSender eventSender) {
+        super(eventSender);
+    }
+
+    @Override
+    public String selector() {
+        return DATALAKE_HORIZONTAL_SCALE_CM_ROLLING_RESTART_HANDLER.selector();
+    }
+
+    @Override
+    public void accept(Event<DatalakeHorizontalScaleFlowEvent> event) {
+        DatalakeHorizontalScaleFlowEvent data = event.getData();
+        stackV4Endpoint.rollingRestartServices(data.getResourceCrn());
+        DatalakeHorizontalScaleFlowEventBuilder resultEventBuilder = DatalakeHorizontalScaleFlowEvent
+                .datalakeHorizontalScaleFlowEventBuilderFactory(data)
+                .setSelector(DATALAKE_HORIZONTAL_SCALE_FINISHED_EVENT.selector());
+        eventSender().sendEvent(resultEventBuilder.build(), event.getHeaders());
+    }
+
+    private BigDecimal executeServicesRollingRestart(DatalakeHorizontalScaleFlowEvent data, ClustersResourceApi clustersResourceApi) {
+        BigDecimal commandId;
+        try {
+            ApiCommand apiCommand = clustersResourceApi.refresh(data.getResourceName());
+            commandId = apiCommand.getId();
+        } catch (ApiException e) {
+            LOGGER.error("CM rolling restart api call failed under the datalake horizontal scale with error msg: {}", e.getMessage());
+            throw new RuntimeException(e);
+        }
+        return commandId;
+    }
+
+    private CMPollingResult pollCM(ApiClient apiClient, BigDecimal commandId) {
+        LOGGER.info("Start CM poll for rolling restart under the datalake horizontal scale");
+        CommandsResourceApi commandsResourceApi = clouderaManagerApiFactory.getCommandsResourceApi(apiClient);
+        Instant timeoutLimit = Instant.now().plusSeconds(timeoutSec);
+        Exception exception = null;
+        while (true) {
```

Comment:
This is a strange pattern, you are triggering the rolling restart through CB, but you are polling from DL. I would trigger and poll CB for the susccess of the command.